api/src/decorators/password-validator.decorator.ts

```diff
@@ -0,0 +1,67 @@
+import {
+  ValidatorConstraint,
+  ValidatorConstraintInterface,
+  ValidationArguments,
+} from 'class-validator';
+import * as config from 'config';
+
+@ValidatorConstraint({ name: 'PasswordValidation', async: false })
+export class PasswordValidation implements ValidatorConstraintInterface {
+  validate(password: string): boolean {
+    return (
+      PasswordValidation.minLengthCheck(password) &&
+      PasswordValidation.upperCaseCheck(password) &&
+      PasswordValidation.includeNumericsCheck(password) &&
+      PasswordValidation.includeSpecialCharactersCheck(password)
+    );
+  }
+
+  private static minLengthCheck(password: string): boolean {
+    const minLength: number = +`${config.get('password.minLength')}`;
+    return minLength <= password.length;
+  }
+
+  private static upperCaseCheck(password: string): boolean {
+    const includeUpperCase: boolean =
+      `${config.get('password.includeUpperCase')}`.toLowerCase() === 'true';
+    return !(includeUpperCase && password.toLowerCase() === password);
+  }
+
+  private static includeNumericsCheck(password: string): boolean {
+    const includeNumerics: boolean =
+      `${config.get('password.includeNumerics')}`.toLowerCase() === 'true';
+    return !(includeNumerics && !/\d/.test(password));
+  }
+
+  private static includeSpecialCharactersCheck(password: string): boolean {
+    const includeSpecialCharacters: boolean =
+      `${config.get('password.includeSpecialCharacters')}`.toLowerCase() ===
+      'true';
+
+    return !(includeSpecialCharacters && !/(?=.*[!@#$%^&*])/.test(password));
+  }
+
+  defaultMessage(args: ValidationArguments): string {
+    const password: string = args.value;
+    let message: string = '';
```

Comment:
make message an array of string, append elements to it inside each `if`, and join() in the end